rpcs3/Emu/Cell/Modules/cellSearch.cpp

```diff
@@ -223,11 +242,173 @@ error_code cellSearchStartListSearch(CellSearchListSearchType type, CellSearchSo
 
 	const u32 id = *outSearchId = idm::make<search_object_t>();
 
-	sysutil_register_cb([=](ppu_thread& ppu) -> s32
+	sysutil_register_cb([=, content_map = g_fxo->get<ContentIdMap>()](ppu_thread& ppu) -> s32
 	{
+		auto curr_search = idm::get<search_object_t>(id);
 		vm::var<CellSearchResultParam> resultParam;
 		resultParam->searchId = id;
-		resultParam->resultNum = 0; // TODO
+		resultParam->resultNum = 0; // Set again later
+
+		std::function<void(const std::string&)> searchInFolder = [&, type](const std::string& vpath)
+		{
+			// TODO: this is just a workaround. On a real PS3 the playlists seem to be stored in dev_hdd0/mms/db/metadata_db_hdd
+
+			std::vector<fs::dir_entry> dirs_sorted;
+
+			for (auto&& entry : fs::dir(vfs::get(vpath)))
+			{
+				entry.name = vfs::unescape(entry.name);
+
+				if (entry.is_directory)
+				{
+					if (entry.name == "." || entry.name == "..")
+					{
+						continue; // these dirs are not included in the dir list
+					}
+
+					dirs_sorted.push_back(entry);
+				}
+			}
+
+			// clang-format off
+			std::sort(dirs_sorted.begin(), dirs_sorted.end(), [&](const fs::dir_entry& a, const fs::dir_entry& b) -> bool
+			{
+				switch (sortOrder)
+				{
+				case CELL_SEARCH_SORTORDER_ASCENDING:
+					// Order alphabetically ascending
+					return a.name < b.name;
+				case CELL_SEARCH_SORTORDER_DESCENDING:
+					// Order alphabetically descending
+					return a.name > b.name;
+				default:
+				{
+					return false;
+				}
+				}
+			});
+			// clang-format on
+
+			for (auto&& item : dirs_sorted)
+			{
+				item.name = vfs::unescape(item.name);
+
+				if (item.name == "." || item.name == "..")
+				{
+					continue;
+				}
+
+				if (!item.is_directory)
+				{
+					continue;
+				}
+
+				const std::string item_path(vpath + "/" + item.name);
+
+				// Count files
+				u32 numOfItems = 0;
+				for (auto&& file : fs::dir(vfs::get(item_path)))
+				{
+					file.name = vfs::unescape(file.name);
+
+					if (file.name == "." || file.name == ".." || file.is_directory)
+					{
+						continue;
+					}
+
+					numOfItems++;
+				}
+
+				const u64 hash = std::hash<std::string>()(item_path);
+				auto found = content_map->find(hash);
+				if (found == content_map->end()) // content isn't yet being tracked
+				{
+					auto ext_offset = item.name.find_last_of('.'); // used later if no "Title" found
+
+					std::shared_ptr<search_content_t> curr_find = std::make_shared<search_content_t>();
+					if (item_path.length() > CELL_SEARCH_PATH_LEN_MAX)
+					{
+						// TODO: Create mapping which will be resolved to an actual hard link in VFS by cellSearchPrepareFile
```

Comment:
Any reason this is unimplemented? I don't expect this code to require major alterations for playlists.